elastic/apm-server
# Handle agentcfg and RUM source mapping ES config parsing consistently

## Motivation/summary


Handle agentcfg and RUM source mapping ES config parsing consistently.
* Accept more ES options in agentcfg ES config
* Fully overwrite ES credentials fields when agentcfg or RUM source mapping ES credentials fields are provided

## Checklist




## How to test these changes


- Configure `agent.config.elasticsearch.hosts` when output is not elasticsearch
- Ensure that `agent.config.elasticsearch` and `rum.source_mapping.elasticsearch` credentials fully overwrites `output.elasticsearch` credentials.

## Related issues



Closes #10659
